lib/Statistics/StatisticsOpensRepository.php

```diff
@@ -58,4 +59,54 @@ public function resetSubscribersScoreCalculation() {
       ->setParameter('verified', null)
       ->getQuery()->execute();
   }
+
+  public function recalculateSegmentScore(SegmentEntity $segment): void {
+    $segment->setAverageEngagementScoreUpdatedAt(new \DateTimeImmutable());
+    $dateTime = (new Carbon())->subYear();
+    $newslettersSentCount = $this
+      ->entityManager
+      ->createQueryBuilder()
+      ->select('count(DISTINCT statisticsNewsletter.newsletter)')
+      ->from(StatisticsNewsletterEntity::class, 'statisticsNewsletter')
+      ->join('statisticsNewsletter.subscriber', 'subscriber')
+      ->join('subscriber.subscriberSegments', 'subscriberSegments')
+      ->where('subscriber.status = :subscribed')
+      ->andWhere('subscriberSegments.segment = :segment')
+      ->andWhere('subscriberSegments.status = :subscribed')
+      ->andWhere('subscriber.deletedAt IS NULL')
+      ->andWhere('subscriber.engagementScore IS NOT NULL')
+      ->andWhere('statisticsNewsletter.sentAt > :dateTime')
+      ->setParameter('segment', $segment)
+      ->setParameter('subscribed', SubscriberEntity::STATUS_SUBSCRIBED)
+      ->setParameter('dateTime', $dateTime)
+      ->getQuery()
+      ->getSingleScalarResult();
+    if ($newslettersSentCount < 3) {
+      $this->entityManager->flush();
+      return;
+    }
+    $avgScore = $this
+      ->entityManager
+      ->createQueryBuilder()
+      ->select('avg(DISTINCT subscriber.engagementScore)')
+      ->from(SubscriberEntity::class, 'subscriber')
+      ->join('subscriber.subscriberSegments', 'subscriberSegments')
+      ->where('subscriberSegments.segment = :segment')
+      ->andWhere('subscriber.status = :subscribed')
+      ->andWhere('subscriberSegments.status = :subscribed')
+      ->andWhere('subscriber.engagementScore IS NOT NULL')
+      ->setParameter('segment', $segment)
+      ->setParameter('subscribed', SubscriberEntity::STATUS_SUBSCRIBED)
+      ->getQuery()
+      ->getSingleScalarResult();
+    $segment->setAverageEngagementScore((float)$avgScore);
+    $this->entityManager->flush();
+  }
+
+  public function resetNewslettersScoreCalculation(): void {
+    $this->entityManager->createQueryBuilder()->update(SegmentEntity::class, 's')
+      ->set('s.averageEngagementScoreUpdatedAt', ':verified')
```

Comment:
I'm not sure why the parameter name is `:verified`. It would be nicer to name it `:updatedAt` or similar so that it doesn't make other devs think about it.